### ISSUE TITLE
Add REST endpoint for network fee recommendation

### ISSUE THREAD

**Issue by evias** [nemtech/catapult-rest#98]
Currently it is hard to tell what amount of transaction fee must be included for a transaction to be confirmed in reasonable time.

With this issue, it is proposed to add an endpoint to catapult-rest that will permit to retrieve information about network fees from the node.

Averages and medians can be computed over a period of `maxRollbackBlocks` blocks.

Following information can be returned by this endpoint:

- averageFeeMultiplier = average fee multiplier over period of `maxRollbackBlocks`.
- medianFeeMultiplier = median fee multiplier over period of `maxRollbackBlocks`.
- highestFeeMultiplier = highest fee multiplier over period of `maxRollbackBlocks`.
- lowestFeeMultiplier = lowest fee multiplier over period of `maxRollbackBlocks`.

This endpoint could also include information about **effectively paid fees** over the period of `maxDifficultyBlocks` such as:

- Effective rental fee to register/extend a namespace per block.
- Effective rental fee to create a subnamespace.
- Effective rental fee to create a mosaic.

**Comment by Vektrat** [comment]
by @dgarcia360 
> We need at least a REST endpoint to return this information.
> It can also include:
> - The median could be computed over last "maxRollBacksBlocks".
> -    Effective rental fee to register/extend a namespace per block.
> - Effective rental fee to create a subnamespace.
> - Effective rental fee to create a mosaic. (edited) 
> 

is the above endpoint "approved", David? you asked if there's any issue security-wise, i'm not sure why would there be one? about other issues we will see when we get into details (calculation performance)

can you provide a suggested endpoint for this?

**Comment by dgarcia360** [comment]
Jag mentioned that this functionality should be available somewhere. It could be placed in a separate microservice or in a new REST endpoint.

I'd advocate for the latest option (REST). We will have more availability rather than trusting in a few microservices instances. 

Regarding security issues, there has not been any major concern until now. My concern was if the response could be verified from client side, but seems so.

I'll provide an initial detailed specification 👍

**Comment by Vektrat** [comment]
for this to happen, we would need to fetch fee information from the core node somehow

@Jaguar0625 @BloodyRookie @gimer

**Comment by dgarcia360** [comment]
1) To calculate ``averageFeeMultiplier``, you have to calculate the average network multiplier over last [maxRollBackBlocks](https://github.com/nemtech/catapult-server/blob/master/resources/config-network.properties#L21) :

1. Query the latest ``maxRollBackBlocks``.
2. For every block, sum the ``feeMultiplier``.
3. Do the average: sum of  feeMultipliers / maxRollBackBlocks

At the same time, you can calculate the median, the highest, and lowest fee multipliers.

2) To retrieve the current effective rental fee for mosaics and namespaces, in my opinion, it would be better if this is returned to you **by the server** since it should be already calculated to determine how much it cost to register a namespace/mosaic.

If not, this is how it's calculated for [mosaics](https://nemtech.github.io/concepts/mosaic.html#rental-fee) and [namespaces](https://nemtech.github.io/concepts/namespace.html#namespace-rental-fee).

**Comment by dgarcia360** [comment]
**Endpoint suggestion:** https://github.com/nemtech/nem2-openapi/pull/55

**Comment by Vektrat** [comment]
i believe for this we need to fetch the `maxRollBackBlocks` somehow... greg will ask core team if there's any way to get it (probably related to node config retrieval)

**Comment by Vektrat** [comment]
we will start working on this because we can build this information with what we currently have in hand

- add a new configuration param will be added at the settings to customize how many blocks should be taken into account to calculate this
- we will fetch and operate on `block.feeMultiplier`
- the suggested endpoint by @dgarcia360 will be used, pending on some late things to take into consideration i suggested: distinction between computed values (stats) and untouched values (properties/settings), otherwise i believe users may get confused

a new issue addressing the conflictive side of this issue will be created (rental fees)
